folding-schemes/src/folding/nova/cyclefold.rs

```diff
@@ -82,6 +87,60 @@ where
     }
 }
 
+impl<C, GC> ToConstraintFieldGadget<CF2<C>> for CycleFoldCommittedInstanceVar<C, GC>
+where
+    C: CurveGroup,
+    GC: CurveVar<C, CF2<C>> + ToConstraintFieldGadget<CF2<C>>,
+    <C as ark_ec::CurveGroup>::BaseField: ark_ff::PrimeField + Absorb,
+    for<'a> &'a GC: GroupOpsBounds<'a, C, GC>,
+{
+    // Extract the underlying field elements from `CycleFoldCommittedInstanceVar`, in the order of
+    // `u`, `x`, `cmE.x`, `cmE.y`, `cmW.x`, `cmW.y`, `cmE.is_inf || cmW.is_inf` (|| is for concat).
+    fn to_constraint_field(&self) -> Result<Vec<FpVar<CF2<C>>>, SynthesisError> {
+        let mut cmE_elems = self.cmE.to_constraint_field()?;
+        let mut cmW_elems = self.cmW.to_constraint_field()?;
+
+        let cmE_is_inf = cmE_elems.pop().unwrap();
+        let cmW_is_inf = cmW_elems.pop().unwrap();
+        // Concatenate `cmE_is_inf` and `cmW_is_inf` to save constraints for CRHGadget::evaluate
+        let is_inf = cmE_is_inf.double()? + cmW_is_inf;
+
+        Ok([
+            nonnative_field_var_to_constraint_field(&self.u)?,
+            self.x
+                .iter()
+                .flat_map(|i| nonnative_field_var_to_constraint_field(i).unwrap())
```

Comment:
I think gracefully handling the `SynthesisError`, if any, might be better here no?